contrib_bots/lib/sched.py

```diff
@@ -0,0 +1,209 @@
+# pip install schedule is necessary
+import schedule
+import time
+import simplejson as json
+import threading
+from random import randint
+
+with open('sched.conf') as data_file:
+    wakeup = json.load(data_file)[0]
+
+# Verify is hour format is valid
+def is_time_format(format):
+    try:
+        time.strptime(format, '%H:%M')
+        return True
+    except ValueError:
+        return False
+
+# For every task we want to schedule we define a job that have a random unique id  to cancel the task.
+
+def job(randomint, message, client):
+    original_content = message['content']
+    reduced_message_list = original_content.split()[3:]
+    reduced_message_string = " ".join(reduced_message_list)
+    if reduced_message_string:
+        if reduced_message_string.split()[0] == "at":
+            reduced_message_string = reduced_message_string.split()[2:]
+            reduced_message_string = " ".join(reduced_message_string)
+
+    events = dict(
+        type='stream',
+        to='alert',
+        subject="All alerts",
+        content=reduced_message_string+" "+randomint,
+        )
+    client.send_message(events)
+
+def start_up(client):
+    events = dict(
+        type='stream',
+        to='alert',
+        subject="All alerts",
+        content="@recover 0",
+        )
+    client.send_message(events)
+    return schedule.CancelJob
+
+class SchedHandler(object):
+    '''
+    This bot allows you to schedule daily alerts. It
+    looks for messages starting with '@every' or '@cancel' or '@recover'.
+
+    In this example, we write alerts at certain time intervals
+    in a new stream called "alert,".
+    '''
+
+    def usage(self, client, state_handler):
```

Comment:
The `usage` method should only responsible for printing usage information:

    $ ./run.py lib/sched.py --config-file ~/w/zuliprc/femnad-zulipdev_cordelia.zuliprc
    Traceback (most recent call last):
      File "./run.py", line 115, in <module>
        run()
      File "./run.py", line 111, in run
        quiet=options.quiet
      File "./run.py", line 59, in run_message_handler_for_bot
        print(message_handler.usage())
    TypeError: usage() missing 2 required positional arguments: 'client' and 'state_handler'